units/patch_intent.go

```diff
@@ -216,11 +217,33 @@ func (j *patchIntentProcessor) finishPatch(patchDoc *patch.Patch, githubOauthTok
 	return nil
 }
 
-func fetchDiffByURL(URL string) (string, error) {
-	client := util.GetHttpClient()
-	defer util.PutHttpClient(client)
+// buildPatchURL creates a URL to enable downloading patch files through the
+// Github API
+func buildPatchURL(gp *patch.GithubPatch) string {
+	url := &url.URL{
+		Scheme: "https",
+		Host:   "api.github.com",
+		Path: fmt.Sprintf("/repos/%s/%s/pulls/%d.diff", gp.BaseOwner,
+			gp.BaseRepo, gp.PRNumber),
+	}
+
+	return url.String()
+}
+
+func fetchDiffByURL(gh *patch.GithubPatch, token string) (string, error) {
```

Comment:
check that gh is not nil

Reply:
GithubPatchData in patches is never nil, so i think it's fair to skip this.